### PR TITLE
Mario/2670 add max amount

### LINKED ISSUE #2670: add "max" button to amount fields
**Is your feature request related to a problem? Please describe.**

Often, users want to do make a tx with all the atoms. since there are 6 decimal places, this can be tedious to type out.

**Describe the solution you'd like**

It is common in these situations to include a `max` button, which will automagically add the total amount left to the amount field for the user.

**Additional context**

Something to consider is the relationship with #2674.

### PR DESCRIPTION
**Pull request by mariopino**
Closes #2670

**Description:**

Add max button to amount field in transfer, delegate and undelegate modals.

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
So this is missing the warning if there is no more funds for fees left as discussed in the issue

**Comment by jbibla** [comment]
i might propose we add this button to `TmField` with a prop `maxAmount` indicating whether or not it should be shown and the amount. 

this would mean we wouldn't have to duplicate the HTML.

do you think this would be a good approach?

**Comment by mariopino** [comment]
Added message:

Warning! you should probably leave some left over for fees

File: src/ActionModal/components/SendModal.vue [hunk 1]
```diff
@@ -91,6 +98,9 @@
         name="Amount"
         type="between"
       />
+      <p v-if="isMaxAmount()" class="form-message notice">
+        Warning! you should probably leave some left over for fees
```
Comment (jbibla) [inline src/ActionModal/components/SendModal.vue]:
```suggestion
        You are about to use all your tokens for this transaction. Consider leaving a little bit left over to cover the network fees.
```

**Comment by colw** [comment]
> i might propose we add this button to `TmField` with a prop `maxAmount` indicating whether or not it should be shown and the amount.
> 
> this would mean we wouldn't have to duplicate the HTML.
> 
> do you think this would be a good approach?

Good suggestion.

Looks great, I've no other comments apart from that.

**Comment by faboweb** [comment]
> i might propose we add this button to TmField with a prop maxAmount indicating whether or not it should be shown and the amount.

TmField is already super cluttered. If we want to do this, let's create another component AmountInput or similar which has limited functionality. We thought about this here #1885. I propose we don't do the refactor here and I reopen the according issue.

**Pull request merged**